### PR TITLE
dockerfile: use moby-bin and cli-bin images for docker binaries

### PR DESCRIPTION
**Pull request by crazy-max**
picked from #2307 

we can now use `moby/moby-bin` and `dockereng/cli-bin` images for testing docker in our integration tests.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -20,26 +20,11 @@ ENV GOFLAGS=-mod=vendor
 ENV CGO_ENABLED=0
 WORKDIR /src
 
+FROM moby/moby-bin:$DOCKER_VERSION AS docker-engine
+FROM dockereng/cli-bin:$DOCKER_VERSION AS docker-cli
```
Comment (thaJeztah) [inline Dockerfile:24]: Do you think there's a need to call out that these are automated builds, not "official releases" (for the occasional reader)?


(We should also really fix those repositories to have a readme to outline their purpose / give the right expectations 😞 - let me see if someone can work on that)

Comment (crazy-max) [reply Dockerfile]: Added "for testing" comment on args if that sgty.

**Review by thaJeztah — APPROVED** [review]
LGTM, yes, that comment should do for now. I created an internal ticket to look at adding README's to those repositories.

File: Dockerfile [hunk 2]
```diff
@@ -3,15 +3,18 @@
 ARG GO_VERSION=1.21
 ARG XX_VERSION=1.4.0
 
+# for testing
 ARG DOCKER_VERSION=25.0.2
 ARG GOTESTSUM_VERSION=v1.9.0
 ARG REGISTRY_VERSION=2.8.0
 ARG BUILDKIT_VERSION=v0.12.5
```
Comment (thaJeztah) [inline Dockerfile:10]: Looks like we have some versions to update here (probably fine to be done separate)

Comment (crazy-max) [reply Dockerfile:10]: Yes done in https://github.com/docker/buildx/pull/2322

**Pull request merged**